superagi/jobs/agent_executor.py

```diff
@@ -292,3 +305,41 @@ def handle_wait_for_permission(self, agent_execution, spawned_agent, session):
         session.add(agent_execution_feed)
         agent_execution.status = "RUNNING"
         session.commit()
+
+    def generate_resource_summary(self, agent_id: int, session: Session, openai_api_key: str):
```

Comment:
Create a new class ResourceSummary and move this method there. Break it into multiple methods.